csi/src/server.rs

```diff
@@ -191,16 +214,36 @@ async fn main() -> Result<(), String> {
         }
     }
 
-    let mut uds_sock = UnixListener::bind(csi_socket).unwrap();
-    info!("CSI plugin bound to {}", csi_socket);
-
-    let uds = Server::builder()
-        .add_service(NodeServer::new(Node {
-            node_name: node_name.into(),
-            filesystems: probe_filesystems(),
-        }))
-        .add_service(IdentityServer::new(Identity {}))
-        .serve_with_incoming(uds_sock.incoming().map_ok(UnixStream));
-    let _ = uds.await;
+    let fut_csi = CSIServer::run(csi_socket, node_name);
+    let fut_grpc =
+        MayastorNodePluginGrpcServer::run(grpc_endpoint(endpoint.to_string()));
+    let futs = join(fut_csi, fut_grpc);
+    let mut rt = Runtime::new().expect("failed to obtain a new Runtime object");
```

Comment:
So its not clear what type of runtime we are building here. Please check the docs, because let us say I have a 96 core server, we might now have 96 threads ... a little much too much, don't you agree? So you can specify what you want exactly i.e:

let mut rt = Runtime::Builder::new().basic_scheduler().enable_io().whatever().build().unwrap();
rt_block_on( ....) 

its not clear though -- as to why you removed tokio::main?

Reply:
Fixed, used tokio::main and tokio::join